app/models/form.rb

```diff
@@ -3,19 +3,30 @@ class Form
               :start_date, :new_logs_end_date, :submission_deadline, :type, :name, :setup_definition,
               :setup_sections, :form_sections, :unresolved_log_redirect_page_id, :edit_end_date
 
+  DEADLINES = {
+    2022 => {
+      submission_deadline: Time.zone.local(2023, 6, 9),
+      new_logs_end_date: Time.zone.local(2023, 11, 20),
+      edit_end_date: Time.zone.local(2023, 11, 20),
+    },
+    2023 => {
+      submission_deadline: Time.zone.local(2024, 6, 7),
+    },
+    2024 => {
+      submission_deadline: Time.zone.local(2025, 6, 6),
+    },
+    :default => {
+      submission_deadline: ->(start_year) { Time.zone.local(start_year + 1, 6, 1) },
```

Comment:
I figured for the displayed submission deadline it was better to have a slightly-early default than anything else.

Reply:
Love this change! Was hoping we will outgrow all the if statements, but now that we're keeping an extra form this is a great refactor!

Reply:
🥳